### PR TITLE
Updated docs for the test refactor (Tests PR 2/3)

### PR DESCRIPTION
**Pull request by huguesdevimeux**
## Changelog / Overview

Updated tests docs.

## Motivation

Following the test refactor (see #1802), the docs have been updated.

### CODE REVIEW

File: docs/source/contributing/testing.rst [hunk 1]
```diff
@@ -167,69 +166,44 @@ For example, to test the ``Circle`` VMobject which resides in
 ``manim/mobject/geometry.py``, add the CircleTest to
 ``test/test_geometry.py``.
 
-In ``test_geometry.py``, add:
-
-.. code:: python
+You will need to use the ``frames_comparison`` decorator to create a test. The test function **must** take as a parameter a variable named ``scene`` that will be used like ``self`` in a standard ``construct`` method.
```
Comment (hydrobeam) [inline docs/source/contributing/testing.rst:169]: This seems important enough that it should be an admonition.

+a clarity suggestion
```suggestion
.. important::
    You will need to use the ``frames_comparison`` decorator to create a test. The test function **must** accept a 
    parameter named ``scene`` that will be used like ``self`` in a standard ``construct`` method.
```

File: docs/source/contributing/testing.rst [hunk 2]
```diff
@@ -167,69 +166,44 @@ For example, to test the ``Circle`` VMobject which resides in
...
-        def construct(self):
-            circle = Circle()
-            self.play(Animation(circle))
+Example : in ``test_geometry.py`` :
 
-Scene names follow the syntax: ``<thing_to_test>Test``. In the example above,
-we are testing whether Circle properly shows up with the generic
-``Animation`` and not any specific animation.
+.. code:: python
+  @frames_comparison
```
Comment (hydrobeam) [inline docs/source/contributing/testing.rst:174]: currently isn't showing up in the docs, needs a newline
```suggestion
.. code:: python

  @frames_comparison
```

File: docs/source/contributing/testing.rst [hunk 3]
```diff
@@ -167,69 +166,44 @@ For example, to test the ``Circle`` VMobject which resides in
...
 
-.. note:: 
+The decorator can be used with or without parentheses. **By default, the test only tests the last frame. To enable multi-frame testing, you have to set ``last_frame=False`` in the parameters.**.
 
-   If the file already exists, edit it and add the test within the file. The 
-   ``Scene`` will be tested thanks to the ``GraphicalUnitTester`` that lives
-   in ``tests/utils/GraphicalUnitTester.py``. Import it with ``from
-   ..utils.GraphicalUnitTester import GraphicalUnitTester``.
+.. code:: python
+  @frames_comparison(last_frame=False)
```
Comment (hydrobeam) [inline docs/source/contributing/testing.rst:182]: same as above
```suggestion
.. code:: python

  @frames_comparison(last_frame=False)
```

File: docs/source/contributing/testing.rst [hunk 4]
```diff
@@ -167,69 +166,44 @@ For example, to test the ``Circle`` VMobject which resides in
...
+  def test_circle(scene):
+      circle = Circle()
+      scene.play(Animation(circle))
 
-To test all the scenes in the module, we do the following:
+You can also specify, when needed, which base scene you need (ThreeDScene, for example) : 
 
 .. code:: python
+  @frames_comparison(last_frame=False, base_scene=ThreeDScene)
```
Comment (hydrobeam) [inline docs/source/contributing/testing.rst:190]: same as above
```suggestion

  @frames_comparison(last_frame=False, base_scene=ThreeDScene)
```

File: docs/source/contributing/testing.rst [hunk 5]
```diff
@@ -167,69 +166,44 @@ For example, to test the ``Circle`` VMobject which resides in
...
-
-    set_test_scene(CircleTest, "geometry")
-
-``set_test_scene`` takes two parameters: the scene to test, and the
-module name. You can generate the test data by running the file (it suffices to type the name of the file in the terminal; you do not have to run
-it like how you would normally run manim files). It will automatically generate the control data in the
-right directory (in this case,
-``tests/control_data/graphical_units_data/geometry/CircleTest.npz``).
+To create the control data for your test, you have to use the flag ``--set_test`` along with pytest. 
+For the example above, it would be ``pytest test_geometry.py::test_circle --set_test -s``. (``-s`` is here to see manim logs, so you can see what's going on).
```
Comment (hydrobeam) [inline docs/source/contributing/testing.rst:204]: I think having having a `.. code-block::bash` with the input needed would make this more explicit/noticeable

**Review by hydrobeam — CHANGES_REQUESTED** [review]
Left some clarity comments + adding some admonitions. Also, for the code examples, I think the import: 

```py
from tests.test_graphical_units.testing.frames_comparison import frames_comparison
```

should be included so that it can just be copy-pasted when writing a new test.

File: docs/source/contributing/testing.rst (hunk 5 above)
Comment (huguesdevimeux) [reply docs/source/contributing/testing.rst:204]: Will do, thanks

File: docs/source/contributing/testing.rst [hunk 6]
```diff
@@ -167,69 +166,59 @@ For example, to test the ``Circle`` VMobject which resides in
...
-.. code:: python
-
-    class CircleTest(Scene):
-        def construct(self):
-            circle = Circle()
-            self.play(Animation(circle))
+To create the control data for your test, you have to use the flag ``--set_test`` along with pytest. 
+For the example above, it would be 
 
+.. code-block::bash
```
Comment (hydrobeam) [inline docs/source/contributing/testing.rst:215]: a space is needed here
```suggestion
.. code-block:: bash
```

File: docs/source/contributing/testing.rst [hunk 7]
```diff
@@ -167,69 +166,59 @@ For example, to test the ``Circle`` VMobject which resides in
...
+The name of the module is indicated by the variable __module_test__, that **must** be decalared in any graphical test file. The module name is used to store the graphical control data.
+
+.. important::
+    You will need to use the ``frames_comparison`` decorator to create a test. The test function **must** accept a 
+    parameter named ``scene`` that will be used like ``self`` in a standard ``construct`` method.
+
+Example : in ``test_geometry.py`` :
 
 .. code:: python
+  from tests.test_graphical_units.testing.frames_comparison import frames_comparison
```
Comment (hydrobeam) [inline docs/source/contributing/testing.rst:178]: A space is needed. Also for the sake of having one complete example, all the imports + `__module_test__`.
```suggestion
.. code:: python
  
  from manim import *
  from tests.test_graphical_units.testing.frames_comparison import frames_comparison

  __module_test__ = "geometry"
  
```

File: docs/source/contributing/testing.rst [hunk 8]
```diff
@@ -167,69 +166,59 @@ For example, to test the ``Circle`` VMobject which resides in
...
-Scene names follow the syntax: ``<thing_to_test>Test``. In the example above,
-we are testing whether Circle properly shows up with the generic
-``Animation`` and not any specific animation.
+  @frames_comparison
+  def test_circle(scene):
+      circle = Circle()
+      scene.play(Animation(circle))
 
-.. note:: 
+The decorator can be used with or without parentheses. **By default, the test only tests the last frame. To enable multi-frame testing, you have to set ``last_frame=False`` in the parameters.**.
```
Comment (hydrobeam) [inline docs/source/contributing/testing.rst:186]: Removed double period.
```suggestion
The decorator can be used with or without parentheses. **By default, the test only tests the last frame. To enable multi-frame testing, you have to set ``last_frame=False`` in the parameters.**
```

File: docs/source/contributing/testing.rst [hunk 9]
```diff
@@ -167,69 +166,59 @@ For example, to test the ``Circle`` VMobject which resides in
 ``manim/mobject/geometry.py``, add the CircleTest to
 ``test/test_geometry.py``.
 
-In ``test_geometry.py``, add:
+The name of the module is indicated by the variable __module_test__, that **must** be decalared in any graphical test file. The module name is used to store the graphical control data.
```
Comment (hydrobeam) [inline docs/source/contributing/testing.rst:169]:
```suggestion
The name of the module is indicated by the variable __module_test__, that **must** be declared in any graphical test file. The module name is used to store the graphical control data.
```

**Review by hydrobeam — CHANGES_REQUESTED** [review]

**Comment by huguesdevimeux** [comment]
@hydrobeam Thanks a lot for your proof-reading !

File: docs/source/contributing/testing.rst [hunk 10]
```diff
@@ -167,69 +166,62 @@ For example, to test the ``Circle`` VMobject which resides in
...
-In ``test_geometry.py``, add:
+The name of the module is indicated by the variable __module_test__, that **must** be declared in any graphical test file. The module name is used to store the graphical control data.
+
+.. important::
+    You will need to use the ``frames_comparison`` decorator to create a test. The test function **must** accept a 
+    parameter named ``scene`` that will be used like ``self`` in a standard ``construct`` method.
+
+Example : in ``test_geometry.py`` :
 
 .. code:: python
```
Comment (hydrobeam) [inline docs/source/contributing/testing.rst:177]:
```suggestion
.. code:: python

```

File: docs/source/contributing/testing.rst [hunk 11]
```diff
@@ -167,69 +166,62 @@ For example, to test the ``Circle`` VMobject which resides in
...
 ``test/test_geometry.py``.
 
-In ``test_geometry.py``, add:
+The name of the module is indicated by the variable __module_test__, that **must** be declared in any graphical test file. The module name is used to store the graphical control data.
+
+.. important::
+    You will need to use the ``frames_comparison`` decorator to create a test. The test function **must** accept a 
+    parameter named ``scene`` that will be used like ``self`` in a standard ``construct`` method.
+
+Example : in ``test_geometry.py`` :
```
Comment (hydrobeam) [inline docs/source/contributing/testing.rst:175]:
```suggestion
Here's an example in ``test_geometry.py``:
```

**Review by hydrobeam — APPROVED** [review]
Final minor suggestions, but LGTM 👍

**Comment by hydrobeam** [comment]
Also, there are some references to the docs for `frames_comparsion` here, but these don't actually exist 🤔, only in file.

**Review by behackl — APPROVED** [review]

**Pull request merged**